src/dependency-manager/src/manager.ts

```diff
@@ -447,10 +450,18 @@ export default abstract class DependencyManager {
     return context;
   }
 
-  async interpolateComponents(graph: DependencyGraph, interpolated_environment: EnvironmentConfig, component_map: Dictionary<ComponentConfig>) {
+  async interpolateComponents(graph: DependencyGraph, interpolated_environment: EnvironmentConfig, component_map: Dictionary<ComponentConfig>, values_dictionary: { [s: string]: { [s: string]: string } }) {
     const environment_components = interpolated_environment.getComponents();
     // Prefix interpolation expressions with components.<name>.
     const prefixed_component_map: Dictionary<ComponentConfig> = {};
+
+    // pre-sort values dictionary to properly stack/override any colliding keys
+    const sorted_values_keys = Object.keys(this.values_dictionary).sort();
```

Comment:
I don't understand this bit. The order of the top level keys in the values.yml matters. They should get applied top down.

Can you add a test to ensure we are respecting order?

Something like:
```yaml
'examples/react-app:*':
  log_level: warn
'*':
  log_level: override_all
```